app/src/org/commcare/android/adapters/HomeScreenAdapter.java

```diff
@@ -0,0 +1,186 @@
+    }};
+
+    //endregion
+
+    //region Private variables
+
+    final View.OnClickListener[] buttonListeners = new View.OnClickListener[buttonsResources.length];
+
+    final SquareButtonWithNotification[] buttons = new SquareButtonWithNotification[buttonsResources.length];
+
+    private Context context;
+
+    private boolean[] hiddenButtons = new boolean[buttonsResources.length];
+    private boolean isInitialized = false;
+
+    private LinkedList<SquareButtonWithNotification> visibleButtons;
+
+    //endregion
+
+    //region Constructors
+
+    public HomeScreenAdapter(Context c) { this.context = c; }
+
+    //endregion
+
+    //region Public API
+
+    public void setOnClickListenerForButton(int androidCode, boolean lookupID, View.OnClickListener listener){
+        int buttonIndex = getButtonIndex(androidCode, lookupID);
+        setOnClickListenerForButton(buttonIndex, listener);
+    }
+
+    public SquareButtonWithNotification getButton(int androidCode, boolean lookupID){
+        return buttons[getButtonIndex(androidCode, lookupID)];
+    }
+
+    public void setNotificationTextForButton(int androidCode, boolean lookupID, String notificationText) {
+        SquareButtonWithNotification button = getButton(androidCode, lookupID);
+        if (button != null) {
+            button.setNotificationText(notificationText);
+            notifyDataSetChanged();
+        }
+    }
+
+    @Override
+    public int getCount() {
+//        return buttonsResources.length;
+        return visibleButtons == null ? buttonsResources.length : visibleButtons.size();
+    }
+
+    @Override
+    public Object getItem(int position) {
+        return buttons[position];
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return buttonsResources[position];
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        View view;
+        if(!isInitialized){
+            visibleButtons = new LinkedList<SquareButtonWithNotification>();
+            Log.i("HomeScrnAdpt","Creating all buttons because got a null in position " + position);
+            for (int i = 0; i < buttons.length; i++) {
+                if (buttons[i] != null) continue;
+                SquareButtonWithNotification button = (SquareButtonWithNotification) LayoutInflater.from(context)
+                        .inflate(buttonsResources[i], parent, false);
+                buttons[i] = button;
+                Log.i("HomeScrnAdpt","Added button " + button + "to position " + i);
+
+                View.OnClickListener listener = buttonListeners[i];
+                // creating now, but set a clickListener before, so we'll add it to this button...
+                if(listener != null) {
+                    button.setOnClickListener(listener);
+                    Log.i("HomeScrnAdpt","Added onClickListener " + listener + " to button in position " + i);
+                }
+                if( i == position ) view = button;
```

Comment:
'view' is and used variable. (Please run the linter on your code pre-PR posting)